File: 232-implement-queue-using-stacks/implement-queue-using-stacks.cpp
#include<bits/stdc++.h>
class MyQueue {
public:
    stack<int> s1,s2;
    int size;
    MyQueue() {
        size = 0;
    }
    
    void push(int x) {
        s1.push(x);
    }
    
    int pop() {
        if(s2.empty()){
            while(!s1.empty()){
                int topE = s1.top();
                s1.pop();
                s2.push(topE);
            }
            int topS2 = s2.top();
            s2.pop();
            return topS2;
        } else{
            int topS2 = s2.top();
            s2.pop();
            return topS2;
        }
    }
    
    int peek() {
        if(s2.empty()){
            while(!s1.empty()){
                int topE = s1.top();
                s1.pop();
                s2.push(topE);
            }
            int topS2 = s2.top();
            return topS2;
        } else{
            int topS2 = s2.top();
            return topS2;
        }
    }
    
    bool empty() {
        return (s1.empty() && s2.empty());
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 232-implement-queue-using-stacks/implement-queue-using-stacks.java
class MyQueue {

    private Stack<Integer> s1, s2;
    
    public MyQueue() {
        s1 = new Stack();
        s2 = new Stack();
    }
    
    public void push(int x) {
        s1.push(x);
    }
    
    public int pop() {
        if(s2.empty()) {
            while(!s1.empty()){
                s2.push(s1.pop());
            }
            return s2.pop();
        } else {
            return s2.pop();
        }        
    }
    
    public int peek() {
        if(s2.empty()) {
            while(!s1.empty()) {
                s2.push(s1.pop());
            }
            return s2.peek();
        } else {
            return s2.peek();
        }
    }
    
    public boolean empty() {
        return (s1.empty() && s2.empty());
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */